include/mbgl/shaders/heatmap_texture_layer_ubo.hpp

```diff
@@ -0,0 +1,18 @@
+#pragma once
+
+#include <mbgl/shaders/layer_ubo.hpp>
+
+namespace mbgl {
+namespace shaders {
+
+struct alignas(16) HeatmapTextureDrawableUBO {
+    std::array<float, 4 * 4> matrix;
+    std::array<float, 2> world;
+    float opacity;
+    bool overdrawInspector;
```

Comment:
Or use a 1-byte value on the GL side.  The other UBOs use int32 for booleans, but we've found UBO updates in the profiling results, so it might be worth minimizing for bandwidth.

Reply:
GLSL doesn't have a 1 byte scalar type (bool is 32 bits). If you had several boolean values, you could pack 4 of them manually in an `int32_t` though.

Reply:
This is a cpp file, so not related to GLSL. The structure used in GLSL is this:
```
layout (std140) uniform HeatmapTextureDrawableUBO {
    highp mat4 u_matrix;
    highp vec2 u_world;
    highp float u_opacity;
    lowp float pad0_;
};
```

Reply:
Right, it doesn't matter so much here as you're just re-using the existing padding for a Metal-only feature.